src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -1936,6 +1959,11 @@ public override bool MoveNext()
 
             public TResult[] ToArray()
             {
+                if (_count == 0)
```

Comment:
Well, actually, it is, but only due to a bug.  Can you spot it? :smile: (https://github.com/dotnet/corefx/pull/5562/files#diff-95a14d3b4ae14339b623dd95c7537fd0R1923)  I'm fixing that, too.